### PR TITLE
KOGITO-1840 enable endpoint for child process

### PR DESCRIPTION
**Pull request by AjayJagan**
- enabled endpoint for child process instances

### CODE REVIEW

**Review by Sara4994 — APPROVED** [review]

File: packages/management-console/src/components/Molecules/ProcessDescriptor/ProcessDescriptor.tsx [hunk 1]
```diff
@@ -26,16 +33,69 @@ const ProcessDescriptor: React.FC<IOwnProps> = ({ processInstanceData }) => {
...
+                      <Badge>{processInstanceData.businessKey}</Badge>
+                    ) : (
+                      idStringModifier(processInstanceData.id)
+                    )}
+                  </span>
+                </Tooltip>
+              </strong>
+            </div>
+          </Link>
+          {processInstanceData.serviceUrl !== null ? (
```
Comment (cristianonicolai) [inline packages/management-console/src/components/Molecules/ProcessDescriptor/ProcessDescriptor.tsx]: @AjayJagan I'm not sure we should combine these into the same component. The descriptor is related to the id and this the endpoint url, how about creating a new component for the endpoint?

**Review by cristianonicolai — COMMENTED** [review]
@AjayJagan there are quite a lot of formatting changes, are those actually expected?

File: packages/management-console/src/components/Molecules/ProcessDescriptor/ProcessDescriptor.tsx (hunk 1 above)
Comment (AjayJagan) [reply packages/management-console/src/components/Molecules/ProcessDescriptor/ProcessDescriptor.tsx]: @cristianonicolai - is it ok to create a separate component just for a link?

Comment (AjayJagan) [reply packages/management-console/src/components/Molecules/ProcessDescriptor/ProcessDescriptor.tsx]: @cristianonicolai - created an endpoint link component for the endpoints separately

Comment (cristianonicolai) [reply packages/management-console/src/components/Molecules/ProcessDescriptor/ProcessDescriptor.tsx]: yep, makes sense to me

**Comment by cristianonicolai** [comment]
@AjayJagan needs rebase now

**Comment by cristianonicolai** [comment]
@AjayJagan can you take a look: ERROR in /home/cnicolai/git/kogito-apps/packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx(4,3)
      TS6133: 'Button' is declared but its value is never read.

**Comment by srambach** [comment]
I talked with the other UXers and we think it's best not to put anything if there is not an endpoint link. So just the button with icon if there's a valid link; otherwise nothing. No tooltip needed and no disabled link. @AjayJagan

**Comment by AjayJagan** [comment]
> I talked with the other UXers and we think it's best not to put anything if there is not an endpoint link. So just the button with icon if there's a valid link; otherwise nothing. No tooltip needed and no disabled link. @AjayJagan

@srambach - The suggested changes have been made. Please have a look

File: packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx [hunk 2]
```diff
@@ -340,6 +339,7 @@ const DataListItemComponent: React.FC<IOwnProps> = ({
           processInstanceData.serviceUrl !== null ? (
             <DataListCheck
               aria-labelledby={'kie-datalist-item-' + processInstanceData.id}
+              name="width-kie-datalist-item"
```
Comment (srambach) [inline packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx:342]: What does this line do?

File: packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx [hunk 3]
```diff
@@ -353,6 +353,7 @@ const DataListItemComponent: React.FC<IOwnProps> = ({
                   aria-labelledby={
                     'kie-datalist-item-' + processInstanceData.id
                   }
+                  name="width-kie-datalist-item"
```
Comment (srambach) [inline packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx]: What does this line do?

**Comment by srambach** [comment]
This seems to work well (not showing the endpoint link if there is none). 
@cristianonicolai Do we display "Endpoint" or something more descriptive both here and in the details. https://github.com/kiegroup/kogito-apps/pull/185#discussion_r410456670
Originally we were just using "Endpoint" so I'm fine either way but it should be consistent.

**Comment by cristianonicolai** [comment]
> This seems to work well (not showing the endpoint link if there is none).
> @cristianonicolai Do we display "Endpoint" or something more descriptive both here and in the details. [#185 (comment)](https://github.com/kiegroup/kogito-apps/pull/185#discussion_r410456670)
> Originally we were just using "Endpoint" so I'm fine either way but it should be consistent.

@srambach I guess the different is that on details page the text is the actual endpoint url, if we want to be consistent then we would display that in the list as well?

File: packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx (hunk 2 above)
Comment (AjayJagan) [reply packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx:342]: I guess we can remove that ☺️

Comment (AjayJagan) [reply packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx:342]: I guess we can remove that ☺️

**Comment by srambach** [comment]
> > This seems to work well (not showing the endpoint link if there is none).
> > @cristianonicolai Do we display "Endpoint" or something more descriptive both here and in the details. [#185 (comment)](https://github.com/kiegroup/kogito-apps/pull/185#discussion_r410456670)
> > Originally we were just using "Endpoint" so I'm fine either way but it should be consistent.
> 
> @srambach I guess the different is that on details page the text is the actual endpoint url, if we want to be consistent then we would display that in the list as well?

Yes. That's what I'm trying to say - put the same (URL) both places. But I think you had a reason why not to put the URL in the list before?

File: packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx (hunk 2 above)
Comment (srambach) [reply packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx:342]: 👍

**Comment by AjayJagan** [comment]
> > > This seems to work well (not showing the endpoint link if there is none).
> > > @cristianonicolai Do we display "Endpoint" or something more descriptive both here and in the details. [#185 (comment)](https://github.com/kiegroup/kogito-apps/pull/185#discussion_r410456670)
> > > Originally we were just using "Endpoint" so I'm fine either way but it should be consistent.
> > 
> > 
> > @srambach I guess the different is that on details page the text is the actual endpoint url, if we want to be consistent then we would display that in the list as well?
> 
> Yes. That's what I'm trying to say - put the same (URL) both places. But I think you had a reason why not to put the URL in the list before?

@srambach - Now I am showing the URL directly in the process list, which is similar to what is being shown in the details page. Please have a look when you have time :)

**Comment by srambach** [comment]
> > > > This seems to work well (not showing the endpoint link if there is none).
> > > > @cristianonicolai Do we display "Endpoint" or something more descriptive both here and in the details. [#185 (comment)](https://github.com/kiegroup/kogito-apps/pull/185#discussion_r410456670)
> > > > Originally we were just using "Endpoint" so I'm fine either way but it should be consistent.
> > > 
> > > 
> > > @srambach I guess the different is that on details page the text is the actual endpoint url, if we want to be consistent then we would display that in the list as well?
> > 
> > 
> > Yes. That's what I'm trying to say - put the same (URL) both places. But I think you had a reason why not to put the URL in the list before?
> 
> @srambach - Now I am showing the URL directly in the process list, which is similar to what is being shown in the details page. Please have a look when you have time :)

I'm sorry - I think I have to change my mind and go back to just having "Endpoint" and the external link icon as the link here in the list. A very long URL pushes the column way out, and although we could truncate/ellipse it, I think it's better to go back to just "Endpoint" (and as you have it, nothing if there is no link)

Sorry for the back and forth. :-(

**Comment by AjayJagan** [comment]
> > > > > This seems to work well (not showing the endpoint link if there is none).
> > > > > @cristianonicolai Do we display "Endpoint" or something more descriptive both here and in the details. [#185 (comment)](https://github.com/kiegroup/kogito-apps/pull/185#discussion_r410456670)
> > > > > Originally we were just using "Endpoint" so I'm fine either way but it should be consistent.
> > > > 
> > > > 
> > > > @srambach I guess the different is that on details page the text is the actual endpoint url, if we want to be consistent then we would display that in the list as well?
> > > 
> > > 
> > > Yes. That's what I'm trying to say - put the same (URL) both places. But I think you had a reason why not to put the URL in the list before?
> > 
> > 
> > @srambach - Now I am showing the URL directly in the process list, which is similar to what is being shown in the details page. Please have a look when you have time :)
> 
> I'm sorry - I think I have to change my mind and go back to just having "Endpoint" and the external link icon as the link here in the list. A very long URL pushes the column way out, and although we could truncate/ellipse it, I think it's better to go back to just "Endpoint" (and as you have it, nothing if there is no link)
> 
> Sorry for the back and forth. :-(

yep no issues :) . Just added back the changes. Please have a look when you have some time . :+1:

**Review by srambach — APPROVED** [review]
Looks good. 👍

**Review by cristianonicolai — APPROVED** [review]

File: packages/management-console/src/components/Organisms/DomainExplorerTable/DomainExplorerTable.tsx [hunk 4]
```diff
@@ -21,7 +21,7 @@ import {
   SearchIcon,
   ErrorCircleOIcon,
   PausedIcon,
-  HistoryIcon
+  HistoryIcon,
```
Comment (nmirasch) [inline packages/management-console/src/components/Organisms/DomainExplorerTable/DomainExplorerTable.tsx]: @AjayJagan is this change needed?

Comment (AjayJagan) [reply packages/management-console/src/components/Organisms/DomainExplorerTable/DomainExplorerTable.tsx]: yep it can be removed :)

**Review by nmirasch — APPROVED** [review]
